lambda.tf

```
@@ -0,0 +1,255 @@
+
+variable "subnet_ids" {
+  description = "List of subnets to use when running in specific VPC."
+  type        = list(string)
+  default     = null
+}
+
+variable "security_group_ids" {
+  description = "List of security group ids when Lambda Function should run in the VPC."
+  type        = list(string)
+  default     = null
+}
+
+variable "lambda_runtime" {
+  type        = string
+  description = "Runtime environment for Datadog lambda"
+  default     = "python3.7"
+}
+
+variable "tracing_config_mode" {
+  type        = string
+  description = "Can be either PassThrough or Active. If PassThrough, Lambda will only trace the request from an upstream service if it contains a tracing header with 'sampled=1'. If Active, Lambda will respect any tracing header it receives from an upstream service."
+  default     = "PassThrough"
+}
+
+variable "dd_api_key_source" {
+  description = "ARN for retrieving api key. Datadog lambda allows for only one of (kms, asm, ssm) secrets management."
+  type = object({
+    resource   = string
+    identifier = string
+  })
+
+  default = {
+    resource   = ""
+    identifier = ""
+  }
+
+  # Resource can be one of kms, asm, ssm ("" to disable all lambda resources)
+  validation {
+    condition     = can(regex("(kms|asm|ssm)", var.dd_api_key_source.resource)) || var.dd_api_key_source.resource == ""
+    error_message = "Provide one, and only one, ARN for (kms, asm) or name (ssm) to retrieve or decrypt Datadog api key."
+  }
+
+  # Check kms arn format
+  validation {
+    condition     = var.dd_api_key_source.resource == "kms" ? can(regex("arn:aws:kms:.*:key/.*", var.dd_api_key_source.identifier)) : true
+    error_message = "ARN for KMS key does not appear to be valid format (example: arn:aws:kms:us-west-2:111122223333:key/1234abcd-12ab-34cd-56ef-1234567890ab)."
+  }
+
+  # Check asm arn format 
+  validation {
+    condition     = var.dd_api_key_source.resource == "asm" ? can(regex("arn:aws:secretsmanager:.*:secret:.*", var.dd_api_key_source.identifier)) : true
+    error_message = "ARN for AWS Secrets Manager (asm) does not appear to be valid format (example: arn:aws:secretsmanager:us-west-2:111122223333:secret:aes128-1a2b3c)."
+  }
+
+  # Check ssm name format
+  validation {
+    condition     = var.dd_api_key_source.resource == "ssm" ? can(regex("^[a-zA-Z0-9_./-]+$", var.dd_api_key_source.identifier)) : true
+    error_message = "Name for SSM parameter does not appear to be valid format, acceptable characters are a-zA-Z0-9_.- and / to delineate hierarchies."
+  }
+}
+
+variable "dd_api_key_kms_ciphertext_blob" {
+  type        = string
+  description = "CiphertextBlob stored in enironment variable DD_KMS_API_KEY used by the lambda function, along with the kms key, to decrypt Datadog API key."
+  default     = ""
+}
+
+locals {
+  lambda_enable          = var.dd_api_key_source.resource != "" ? true : false
```

```suggestion
  lambda_enabled          = var.dd_api_key_source.resource != "" ? true : false
```

we usually use the suffix `_enabled` for boolean enable/disable flags